### PR TITLE
GitLab recipe

### PR DESCRIPTION
**Pull request by digiguru**
Adds a recipe for publishing to GitLab Pages based on the GitHub template.

### CODE REVIEW

File: docs/gitlab-pages.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Gitlab Pages
+
+You don't have to use Github to serve Foam pages. You can also use Gitlab.
```
Comment (jsjoeio) [inline docs/gitlab-pages.md:3]:
```suggestion
# GitLab Pages

You don't have to use GitHub to serve Foam pages. You can also use GitLab.
```

File: docs/gitlab-pages.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Setup a project
+
+### Generate the directory from github
+
+Generate a solution using the [Foam template].
+
+Change the remote to Gitlab, or copy all the files into a new gitlab repo.
```
Comment (jsjoeio) [inline docs/gitlab-pages.md:11]:
```suggestion
Change the remote to GitLab, or copy all the files into a new GitLab repo.
```

File: docs/gitlab-pages.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+``` 
+
+Commit the file and push it to gitlab.
+
+## Setup CI/CD
+
+From the project home in gitlab click `Set up CI/CD`
+Choose `Jekyll` as your template from the template dropdown
+Click `commit`
+No when you go to CI / CD > Pipelines you should see the code running
```
Comment (jsjoeio) [inline docs/gitlab-pages.md:44]:
```suggestion
1. From the project home in GitLab click `Set up CI/CD`
2. Choose `Jekyll` as your template from the template dropdown
3. Click `commit`
4. Now when you go to CI / CD > Pipelines, you should see the code running
```

File: docs/gitlab-pages.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+## Troubleshooting
+
+- *Could not locate Gemfile* - You didn't follow the steps above to [#Add a Gemlock file]
+- *Conversion error: Jekyll::Converters::Scss encountered an error while converting* You need to reference a theme.
+- *Pages are running in CI/CD, but I only ever see `test`, and never deploy* - Perhaps you've renamed the main branch (from master) - check the settings in `.gitlab-ci.yml` and ensure the deploy command is running to the branch you expect it to.
+- *I deployed, but my .msdfiles don't seem to be being converted into .html files* - You need a gem that Github installs by default - check `gem "jekyll-optional-front-matter"` appears in the `Gemfile`
```
Comment (jsjoeio) [inline docs/gitlab-pages.md:51]:
```suggestion
- *I deployed, but my .msd files don't seem to be being converted into .html files* - You need a gem that GitHub installs by default - check `gem "jekyll-optional-front-matter"` appears in the `Gemfile`
```

Comment (digiguru) [reply docs/gitlab-pages.md:51]: Thanks @jsjoeio

**Comment by digiguru** [comment]
Thanks for the review

File: docs/gitlab-pages.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# GitLab Pages
+
+You don't have to use GitHub to serve Foam pages. You can also use GitLab.
+
+## Setup a project
+
+### Generate the directory from github
```
Comment (jsjoeio) [inline docs/gitlab-pages.md:7]:
```suggestion
### Generate the directory from GitHub
```

**Comment by jevakallio** [comment]
@all-contributors add @digiguru for code,doc 🎉

**Comment by jevakallio** [comment]
@alll-contributors add @digiguru for code and doc 😄

**Comment by jevakallio** [comment]
@all-contributors add @digiguru for code and doc, please 😆

**Comment by jevakallio** [comment]
@all-contributors I can't believe you made me beg for it

**Comment by digiguru** [comment]
lol - don't anger the robots

**Pull request merged**